### PR TITLE
Generate scaladoc 

### PR DESCRIPTION
**Pull request by simerplaha**
## Overview

- PR for [comment](https://github.com/alephium/explorer-backend/pull/165#issuecomment-1066064429)
- Adds plugin `sbt-unidoc` to support multiple projects references in docs.
- References to Slick types link to https://www.javadoc.io/doc/com.typesafe.slick/slick_2.13/3.3.2 
- References to Scala types link to https://scala-lang.org/api/2.13.3

## Generating docs
`sbt unidoc`

See [readme.md](https://github.com/alephium/explorer-backend/blob/c5cb628d684c30eb3d1efd8968f2f3b3bed96ea3/README.md#scaladoc)

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -139,6 +206,16 @@ lazy val app = mainProject("app")
       case "module-info.class" =>
         MergeStrategy.discard
       case other => (assembly / assemblyMergeStrategy).value(other)
+    },
+    apiMappings ++= {
+      val slickAPIMapping =
```
Comment (tdroxler) [inline build.sbt:211]: we could maybe have this `val slickAPIMapping` define outside so we can reuse it here and below

**Review by tdroxler — APPROVED** [review]
Great :rocket: LGTM

**Review by polarker — APPROVED** [review]
Thanks ❤️

File: build.sbt (hunk 1 above)
Comment (simerplaha) [reply build.sbt:211]: Yep that would've been much cleaner. But sbt doesn't allow it. 

> error: `value` can only be called on a task within a task definition macro
classPath    = (Compile / fullClasspath).value

I'm sure there is a way to make this work, it's Scala after all. Will try to figure it out another PR.

**Pull request merged**